docs/csharp/whats-new/csharp-8.md

```diff
@@ -0,0 +1,359 @@
+};
+```
+
+The discard pattern in the preceding switch matches when either `x` or `y`, but not both, is 0. A switch expression must either produce a value or throw an exception. If none of the cases match, the switch expression throws an exception. The compiler generates a warning for you if you do not cover all possible cases in your switch expression.
+
+## using declarations
+
+A **using declaration** is a variable declaration preceded by the `using` keyword. It tells the compiler that the variable being declared should be disposed at the end of the enclosing scope. For example, consider the following code that writes a text file:
+
+```csharp
+static void WriteLinesToFile(IEnumerable<string> lines)
+{
+    using var file = new System.IO.StreamWriter("WriteLines2.txt");
+    foreach (string line in lines)
+    {
+        // If the line doesn't contain the word 'Second', write the line to the file.
+        if (!line.Contains("Second"))
+        {
+            file.WriteLine(line);
+        }
+    }
+// file is disposed here
+}
+```
+
+In the preceding example, the file is disposed when the closing brace for the method is reached. That's the end of the scope in which `file` is declared. The preceding code is equivalent to the following code using the classic [using statements](../language-reference/keywords/using-statement.md) statement:
+
+
+```csharp
+static void WriteLinesToFile(IEnumerable<string> lines)
+{
+    using (var file = new System.IO.StreamWriter("WriteLines2.txt"))
+    {
+        foreach (string line in lines)
+        {
+            // If the line doesn't contain the word 'Second', write the line to the file.
+            if (!line.Contains("Second"))
+            {
+                file.WriteLine(line);
+            }
+        }
+    } // file is disposed here
+}
+```
+
+In the preceding example, the file is disposed when the closing brace associated with the `using` statement is reached.
+
+In both cases, the compiler generates the call to `Dispose()`. The compiler generates an error if the expression in the using statement is not disposable.
+
+## Static local functions
+
+You can now add the `static` modifier to local functions to ensure that local function doesn't capture (reference) any variables from the enclosing scope. Doing so generates `CS8421`, "A static local function can't contain a reference to <variable>." 
+
+Consider the following code. The local function `LocalFunction` accesses the variable `y`, declared in the enclosing scope (the method `M`). Therefore, `LocalFunction` can't be declared with the `static` modifier:
+
+```csharp
+int M()
+{
+    int y;
+    LocalFunction();
+    return y;
+
+    void LocalFunction() => y = 0;
+}
+```
+
+The following code contains a static local function. It can be static because it doesn't access any variables in the enclosing scope:
+
+```csharp
+int M()
+{
+    int y = 5;
+    int x = 7;
+    return Add(x, y);
+
+    int Add(int left, int right) => left + right;
```

Comment:
@BillWagner is the local function intentionally not `static` in this example, though it can be such? For the sake of example, adding `static` modifier is reasonable.